File: src/main/java/app/application/utils/service/YoutubeDownloadService.java
package app.application.utils.service;


import app.application.listener.YoutubeDownloadListener;
import app.application.utils.DialogManager;
import app.application.utils.UserConfigHandler;
import com.github.kiulian.downloader.YoutubeDownloader;
import lombok.Setter;
import org.springframework.beans.factory.annotation.Autowired;

public abstract class YoutubeDownloadService {

    @Autowired
    protected YoutubeDownloader youtubeDownloader;

    @Autowired
    protected UserConfigHandler userConfigHandler;

    @Autowired
    protected DialogManager dialogManager;

    @Setter
    protected YoutubeDownloadListener youtubeDownloadListener;

}

File: src/main/kotlin/app/application/utils/service/download/YoutubeDownloadService.kt
package app.application.utils.service.download

import app.application.listener.YoutubeDownloadListener
import app.application.utils.DialogManager
import app.application.utils.UserConfigHandler
import app.application.utils.converter.YoutubeVideoConverter
import com.github.kiulian.downloader.YoutubeDownloader
import org.springframework.beans.factory.annotation.Autowired

abstract class YoutubeDownloadService{
    @Autowired
    protected lateinit var youtubeDownloader: YoutubeDownloader

    @Autowired
    protected lateinit var userConfigHandler: UserConfigHandler

    @Autowired
    protected lateinit var dialogManager: DialogManager

    var youtubeDownloadListener: YoutubeDownloadListener? = null

    val youtubeVideoConverter = YoutubeVideoConverter()

}